doc/specs/#1595 - Suggestions UI/Suggestions-UI.md

```diff
@@ -0,0 +1,734 @@
+
+### 🏃‍♂️ Run
+
+* [ ] Add a `description` field to `Command`
+* [ ] Add a `TeachingTip` (or similar) to the Suggestions UI to display
+  descriptions (when available)
+* [ ] Use the `ToolTip` property of shell-driven suggestions as the description
+* [ ] Add a boolean `nesting` property which can be used to disable nesting on the `tasks` source.
+* [ ] Add the ability for `nesting` to accept `enabled`/`disabled` as `true`/`false` equivalents
+* [ ] Add the ability for `nesting` to accept `source`, which instead groups all
+  commands to the Suggestions UI by the source of that suggestion. 
+
+### 🚀 Sprint
+
+The two "sprint" tasks here are much more ambitious than the other listed
+scenarios, so breaking them down to atomic tasks sees less reasonable. We'd have
+to spend a considerable amount more time figuring out _how_ to do each of these
+first. 
+
+For example - extensions. We have yet to fully realize what extensions _are_.
+Determining how extensions will provide suggestions is left as something we'll
+need to do as a part of the Extensions spec.
+
+## Conclusion
+
+Here's a sample json schema for the settings discussed here.
+
+```json
+"OpenSuggestionsAction": {
+  "description": "Arguments corresponding to a Open Suggestions Action",
+  "allOf": [
+    {
+      "$ref": "#/$defs/ShortcutAction"
+    },
+    {
+      "properties": {
+        "action": {
+          "type": "string",
+          "const": "suggestions"
+        },
+        "source": {
+          "$ref": "#/$defs/SuggestionSource",
+          "description": "Which suggestion sources to filter."
+        },
+        "useCommandline": {
+          "default": false,
+          "description": "When set to `true`, the current commandline the user has typed will pre-populate the filter of the Suggestions UI. This requires that the user has enabled shell integration in their shell's config. When set to false, the filter will start empty."
+        },
+        "nesting": {
+          "default": true,
+          "description": "When set to `true`, suggestions will follow the provided nesting structure. For Tasks, these will follow the structure of the Command Palette. When set to `false`, no nesting will be used (and all suggestions will be in the top-level menu.",
+          "$comment": "This setting is a possible follow-up setting, not required for v1. "
+        }
+      }
+    }
+  ]
+},
+"SuggestionSource": {
+  "default": "all",
+  "description": "Either a single suggestion source, or an array of sources to concatenate. Built-in sources include `commandHistory`, `directoryHistory`, `tasks`, and `local`. Extensions may provide additional values. The special value `all` indicates all suggestion sources should be included",
+  "$comment": "`tasks` and `local` are sources that would be added by the Tasks feature, as a follow-up"
+  "oneOf": [
+    {
+      "type": [ "string", "null" ]

```

Comment:
can we add the built-in sources somehow? We don't have extensions yet, so we can be exhaustive.

Reply:
maybe. JSON schema is wacky but I think I got it.